src/components/map/FeaturePropertyPopup.jsx

```diff
@@ -21,8 +21,9 @@ function renderProperties(feature) {
   })
 }
 
-function renderFeature(feature) {
-  return <div key={`${feature.sourceLayer}-${feature.id}`}>
+function renderFeature(feature, idx) {
+  const uniqueKey = feature.hasOwnProperty('id') ? feature.id : idx;
```

Comment:
This is a little bit of a hack because ideally you shouldn't use an arrays index for uniqueness in react. However we don't have anything else unique to reference

See <https://github.com/mapbox/mapbox-gl-js/blob/b3046ea99a73cc027aa5d45ddba5285670097af8/src/util/vectortile_to_geojson.js#L8>

Reply:
Just had a look at the spec:

https://github.com/mapbox/vector-tile-spec/tree/master/2.1#42-features

> A feature MAY contain an id field. If a feature has an id field, the value of the id SHOULD be unique among the features of the parent layer.

So if there are feature IDs and they are not unique (there is only _should_ in the spec) the issue would still exist, right?

What about using only the index here?

Reply:
Good spot @pathmapper